tests/test_blob_repairer.py

```diff
@@ -0,0 +1,434 @@
+    header = make_header(flags=RecordFlags(flags), disk_size=128)
+
+    blob_repairer = BlobRepairer(TEST_BLOB_PATH)
+    assert blob_repairer.check_header(header)
+
+
+@mock.patch('eblob_kit.IndexFile', autospec=True)
+@mock.patch('eblob_kit.DataFile', autospec=True)
+@mock.patch('os.path.exists', return_value=True)
+def test_check_header_with_exthdr_and_datasize(_mocked_exists, mocked_data_file, _index_file):
+    """Check header with EXTHDR and incorrect datasize."""
+    header = make_header(data_size=128, disk_size=128, flags=RecordFlags(RecordFlags.EXTHDR))
+
+    mocked_data_file.return_value.__len__.return_value = 512
+
+    blob_repairer = BlobRepairer(TEST_BLOB_PATH)
+    assert not blob_repairer.check_header(header)
+
+
+@pytest.mark.parametrize('position, end, header_position, holes, holes_size, check_header_called',
+                         [
+                             # position, end, hdr_pos, holes, holes_size, check_header_called
+                             (0,   3,   0, 1, 3, 1),
+                             (0,   1, 100, 1, 1, 1),
+                             (1,   3,   1, 1, 2, 1),
+                             (5,   6,   6, 1, 1, 1),
+                             (5,   5,   5, 0, 0, 0),
+                         ])
+@mock.patch('eblob_kit.IndexFile', autospec=True)
+@mock.patch('eblob_kit.DataFile', autospec=True)
+@mock.patch('os.path.exists', return_value=True)
+def test_check_hole(_mock_exist,
+                    data_mock,
+                    _index_mock,
+                    mocker,
+                    position,
+                    end,
+                    header_position,
+                    holes,
+                    holes_size,
+                    check_header_called):
+    """Check generated DiskControl header with BlobRepairer.check_hole.
+
+    TODO(karapuz): try to move common for all test patch code to BlobRepairer fixture.
+
+    :param int position:
+    :param int end:
+    :param int header_position:
+    :param int holes:
+    :param int holes_size:
+    :param int check_header_called:
+
+    """
+    # DataFile mock
+    dummy_header = make_header(position=header_position)
+    data_mock.return_value.read_disk_control.return_value = dummy_header
+
+    blob_repairer = BlobRepairer(TEST_BLOB_PATH)
+    assert blob_repairer.valid
+
+    mocker.spy(blob_repairer, 'check_header')
+
+    blob_repairer.check_hole(position, end)
+
+    assert blob_repairer.stat.holes == holes
+    assert blob_repairer.stat.holes_size == holes_size
+
+    if check_header_called:
+        blob_repairer.check_header.assert_called_with(dummy_header)
+
+    assert blob_repairer.check_header.call_count == check_header_called
+
+
+@pytest.mark.parametrize('header_index, position, headers', [
+    (1, 8, [make_header(disk_size=5, position=3), empty_header()]),
+    (0, 0, [empty_header(), empty_header()]),
+    (1, 11, [make_header(disk_size=5, position=3), empty_header()]),
+])
+@pytest.mark.xfail(raises=AssertionError)
```

Comment:
`xfail`?

Reply:
Better to stay more specific with exception being thrown as  `resolve_mispositioned_record` uses assert inside.

Reply:
Let's use local `with pytest.raises(AssertionError):` around `resolve_mispositioned_record`, because with current variant any assert will be considered as expected.